lib/diagnostic.js

```diff
@@ -4,31 +4,59 @@
 // Write a function that takes zero or more arguments
 // and returns the minimum from that set
 // assume that all the argument are numbers
-const min = function min() {
+const min = function () {
+  let result = arguments[0];
+  for (let i = 0; i < arguments.length; i++) {
+    if (result > arguments[i]){
+      result = arguments[i];
+    }
+  }
+  return result;};
 
-};
 
 // Question 2
 // Write a function that takes zero or more arguments
 // and returns the sum of that set
 // assume that all the argument are numbers
 const sum = function sum() {
-
+  let result = 0;
+  for (let i = 0; i < arguments.length; i++) {
+    result += arguments[i];
+  }
+return result;
 };
 
 // Question 3
 // Write a function that returns an array of the specified length
 // with all the elements set to defaultValue
 const newArray = function newArray(length, defaultValue) {
-
+  let result = [];
+  for (var i = 0; i < newArray.length; i++) {
+    result.push(defaultValue);
+  }
+  return result;
```

Comment:
the new array should be created in the function `let array = new Array(length);`